app/src/main/java/com/example/cooktaildb/ui/home_fragment/HomeFragment.kt

```diff
@@ -1,9 +1,53 @@
 package com.example.cooktaildb.ui.home_fragment
 
+import android.os.Bundle
+import android.view.View
 import com.example.cooktaildb.base.BaseFragment
+import com.example.cooktaildb.data.model.Category
+import com.example.cooktaildb.data.repository.CategoryRepository
+import com.example.cooktaildb.data.source.remote.RemoteCategoryDataSource
 import com.example.cooktaildb.databinding.FragmentHomeBinding
+import com.example.cooktaildb.ui.home_fragment.adapter.DrinkViewPagerAdapter
+import com.example.cooktaildb.ui.home_fragment.presenter.HomeFragmentContract
+import com.example.cooktaildb.ui.home_fragment.presenter.HomeFragmentPresenter
+import com.google.android.material.tabs.TabLayoutMediator
 
-class HomeFragment : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::inflate) {
+class HomeFragment : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::inflate), HomeFragmentContract.View {
+    private var homePresenter: HomeFragmentPresenter? = null
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        homePresenter = HomeFragmentPresenter(CategoryRepository.getInstance(RemoteCategoryDataSource.getInstance()), this)
+
+        homePresenter?.getCategory()
+    }
+
+    override fun getListCategorySuccess(categories: List<Category>) {
+        binding?.viewpager2Drink?.adapter = DrinkViewPagerAdapter(categories)
+        binding?.tabCategory?.let {
+            binding?.viewpager2Drink?.let { it1 ->
+                TabLayoutMediator(it, it1){ tab, position ->
+                    tab.text = categories[position].strCategory
+                }.attach()
+            }
+        }
```

Comment:
```suggestion
               binding?.apply {
            viewpager2Drink.adapter = DrinkViewPagerAdapter(categories)
            tabCategory?.let {
                viewpager2Drink?.let { it1 ->
                    TabLayoutMediator(it, it1){ tab, position ->
                        tab.text = categories[position].strCategory
                    }.attach()
                }
            }
        }
```it1 đặt lại tên cho trực quan nhé ông